astropy/table/index.py

```diff
@@ -975,22 +1011,58 @@ class TableLoc:
     ----------
     table : Table
         Indexed table to use
+    index_name : tuple or None
+        If not None, the index name as a tuple to use for all retrievals. If None
+        (default), the primary key index is used.
     """
 
-    def __init__(self, table):
+    def __init__(self, table: Table, index_name: tuple | None = None):
         self.table = table
         self.indices = table.indices
+        self.index_name = index_name
+
+    def __call__(self, *index_name):
+        if len(index_name) == 1 and isinstance(index_name[0], (tuple, list)):
```

Comment:
Your example was `t.loc("b")[1]` - that would fail, correct? Was the plan just to pass on `index_name` (which is a tuple)?